sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -487,20 +487,28 @@ string_class ProgramManager::getProgramBuildLog(const RT::PiProgram &Program,
   string_class Log = "The program was built for " +
                      std::to_string(PIDevices.size()) + " devices";
   for (RT::PiDevice &Device : PIDevices) {
+    std::string DeviceBuildInfoString = "";
     Plugin.call<PiApiKind::piProgramGetBuildInfo>(
         Program, Device, CL_PROGRAM_BUILD_LOG, 0, nullptr, &Size);
-    vector_class<char> DeviceBuildInfo(Size);
-    Plugin.call<PiApiKind::piProgramGetBuildInfo>(
-        Program, Device, CL_PROGRAM_BUILD_LOG, Size, DeviceBuildInfo.data(),
-        nullptr);
+    if (Size > 0) {
+      vector_class<char> DeviceBuildInfo(Size);
+      Plugin.call<PiApiKind::piProgramGetBuildInfo>(
+          Program, Device, CL_PROGRAM_BUILD_LOG, Size, DeviceBuildInfo.data(),
+          nullptr);
+      DeviceBuildInfoString = string_class(DeviceBuildInfo.data());
```

Comment:
```suggestion
      DeviceBuildInfoString = std::string(DeviceBuildInfo.data());
```
Since the `..._class` from SYCL 1.2.1 are deprecated.

Reply:
good to know. thanks.